runtime/near-vm-logic/src/gas_counter.rs

```diff
@@ -87,21 +87,24 @@ impl GasCounter {
         }
     }
 
-    /// Accounts for burnt and used gas; reports an error if max gas burnt or
-    /// prepaid gas limit is crossed.  Panics when trying to burn more gas than
-    /// being used, i.e. if `burn_gas > use_gas`.
-    fn deduct_gas(&mut self, burn_gas: Gas, use_gas: Gas) -> Result<()> {
-        assert!(burn_gas <= use_gas);
-        let promise_gas = use_gas - burn_gas;
+    /// Deducts burnt and used gas.
+    ///
+    /// Returns an error if the max_gax_burnt or the prepaid_gas limits are
+    /// crossed or there are arithmetic overflows.
+    ///
+    /// Preconditions: gas_burnt <= gas_used
```

Comment:
```suggestion
    /// Returns an error if the `max_gax_burnt` or the `prepaid_gas` limits are
    /// crossed or there are arithmetic overflows.
    ///
    /// Preconditions: `gas_burnt <= gas_used`
```

As far as panics are concerned the convention is usually something like this in order to make it stand out:

```suggestion
    /// Returns an error if the `max_gax_burnt` or the `prepaid_gas` limits are
    /// crossed or there are arithmetic overflows.
    ///
    /// # Panics
    ///
    /// This function asserts that `gas_burnt <= gas_used`.
```